serverless/handlers/resolve-chats/index.js

```diff
@@ -15,26 +15,76 @@ exports.handler = async (event) => {
   return [];
 };
 
-async function getChats(eventId) {
-  const searchParamsBase = {
+async function getChats(eventId, id) {
+  // get all chats for the provided eventId
+  const eventChats = await getEventChats(eventId);
+  // get ids for matches where the current attendee id is one of the attendees in the match
+  const attendeeMatchIds = await getMatchesForAttendee(eventId, id).then((matches) =>
+    matches.map((match) => match.id)
+  );
+  // filter the event chats for matchIds that exist in the matches array
+  const relevantChats = eventChats.filter((chat) => attendeeMatchIds.includes(chat.matchId));
+  return relevantChats || [];
+}
+
+async function getEventChats(eventId) {
+  const searchParams = {
+    ExpressionAttributeNames: {
+      '#eventId': 'eventId',
+    },
+    ExpressionAttributeValues: {
+      ':eventId': eventId,
+    },
     KeyConditionExpression: '#eventId = :eventId',
     IndexName: 'byEventId',
     TableName: CHAT_THREAD_TABLE_NAME,
   };
 
+  const Items = await documentClient()
+    .query(searchParams)
+    .promise()
+    .then((res) => [...res.Items]);
+
+  return Items.length ? Items : [];
+}
+
+async function getMatchesForAttendee(eventId, attendeeId) {
```

Comment:
nit: we could remove the dupe function here and import the one from `resolve-matches/index.js`.